src/core/microservices/notifications.payload.builder.ts

```diff
@@ -361,16 +353,26 @@ export class NotificationsPayloadBuilder {
           LogContext.CHALLENGES
         );
       }
-
-      payload.hub.challenge = {
-        nameID: await this.getChallengeNameID(parentCommunity.parentID),
-        id: parentCommunity.parentID,
+      challengeNameID =
+        (await this.getChallengeNameID(parentCommunity.parentID)) || '';
+      challengeID = parentCommunity.parentID;
+      opportunityID = community.parentID;
+      opportunityNameID =
+        (await this.getOpportunityNameID(community.parentID)) || '';
+    }
+    const result: HubPayload = {
+      id: community.hubID,
+      nameID: (await this.getHubNameID(community.hubID)) || '',
+      challenge: {
+        id: challengeID,
+        nameID: challengeNameID,
         opportunity: {
-          nameID: await this.getOpportunityNameID(community.parentID),
-          id: community.parentID,
+          id: opportunityID,
+          nameID: opportunityNameID,
         },
-      };
-    }
+      },
+    };
+    return result;
   }
```

Comment:
I think variables here should not default to an empty string.
`HubPayload ` suggests  that if you have challenge or opportunity fields then a value is required.